### DISCUSSION TITLE
Voice chat / mic settings, noise suppression, echo removal

### DISCUSSION THREAD

**Discussion by xMasterX** [stoatchat/discussions#277] [Feature Suggestions]
Revolt really needs to have such features as:

- Proper mic activation settings - Detect voice using [RNNoise](https://gitlab.xiph.org/xiph/rnnoise), Amplitude detection, etc
- Mic Volume
- Noise suppression, for example it can be done with [RNNoise](https://gitlab.xiph.org/xiph/rnnoise)
- Echo removal option
- Voice compression settings, codec settings

Text chats are ok, but for me main purpose of that apps is voice chat, and at the moment best quality only in teamspeak 3, after they removed CELT Mono codec, quality got worse, but not as much as it is in discord, and at least I can select opus codec settings.

Screen share is important too, but I think clean not distorted sound need to be first priority.

There is an example mic settings from Mumble client:
![pic](https://user-images.githubusercontent.com/10697207/145698137-56201cd4-2833-403b-bb90-d27a4d7f062e.jpg)

**Comment by Itsredsect** [comment]
hopefully this wil be implemented in the future

**Comment by Smokemic** [comment]
It´s been a while is there any Option to implement this myself?

**Comment by nulldg** [comment]
+1

while i personally don't use VoIP, i recognize that it's a big selling point for many chat applications. it's not enough these days to simply have VoIP, you gotta do it well too.

**Comment by Sooperfish** [comment]
One of my main gripes with Discord is that Krisp implementation in Discord is ass and keeps turning off. I believe Discord is not able to alter or tweak Krisp a lot for better implementation because Krisp is a third party closed source application that's only integrated under certain terms and conditions. I would love for Revolt to have some open source krisp alternative integrated. 

Other issue I experience is that a lot of users don't know how / put in the effort to change to optimal noise suppression settings, which can be annoying for people sensitive to noises (especially eating!). Perhaps a client side noise suppression for other users (on the output end) or a permission to force microphone noise suppression on users would be nice to have.

**Comment by koloved** [comment]
We tried Revolt few times with my friends when discord not working well and every time we switch to discord cause background noise. the Neural noise suppression, echo removal its the MAIN feature that Revolt should have or Revolt is unusable.

**Reply by Schweeeeeeeeeeeeeeee** [reply]
i agree most defenitely unusabe with background noise removal (breathing, computer sounds, keyboards, mice, lawn mowers, basically anything that isnt a voice)

**Comment by Nekall** [comment]
If it had microphone settings, noise suppression (and window/screen sharing), I'd replace Discord instantly. I hope it will show up soon! But when I see the post's date... I'm scared.

**Comment by klausmyrseth** [comment]
This is readlly needed, the current brtowser noise supression is not working at all neither app nor browser. Its picking up everything like its no noise supression at all.

This is probably one of the most needed function to make a discord replacement, the other one is share screen as a close second but absolutely not more important then noise canceling.

**Comment by Zomatree** [comment]
We are currently evaluating using rnnoise, it's basically the only viable model for noise suppression which is easily implementable for us, Krisp is something we have looked into using but they denied our application to use them due to "existing contractual obligations" (it should be obvious what this is referring to). 

WebRTC implements a default noise suppression algorithm which can also use but as far as I know it's not the best out there.

Unrelated but regarding video streaming, it's implemented and hoped to be released soon.

**Reply by thrasmionOS** [reply]
Let me know if you need any help with coding or integration of rnnoise. Happy to volunteer the time to get these features in sooner.

**Comment by samoshi** [comment]
A +1 on this would be having the settings tab able to open from within a call so you can quickly make changes without going to a different settings menu.

**Comment by SteFre95** [comment]
Something I also noticed while using the Windows desktop app is that it gradually lowers the mic volume in the Windows audio manager while in a voice chat, making me and my friends get quieter over time. Turning it back up works for a short time, until it auto-adjusts down again.
I assume there is some auto volume / gain control at play here, which should be toggleable in the voice settings of the app as to not change peoples' mic settings.

**Reply by BuddaPB** [reply]
same issue on Linux (which has been resolved by forcing the microphone level with Pavucontrol)

**Comment by thrasmionOS** [comment]
I need this and PTT features asap as this is blocking me from getting many of my friends to migrate over off of that garbage heap that is Discord.

**Reply by Loopw4** [reply]
I totally agree. This is what Stoat needs to be a proper Discord alternative. I hope the increased visibility will also bring more developers willing to spend time on the project!

**Comment by klausmyrseth** [comment]
if you use equalizer apo with plugins you can get this done easilly on your own end.

rnnoise as vst plugin and you can get it working ;) thats what im using to use all kinds of services like this.

**Comment by Dadadah** [comment]
To those of you following this issue, an [RNNoise PR](https://github.com/stoatchat/for-web/pull/783) is currently in the review process.

**Reply by thrasmionOS** [reply]
I see that is in the Web client base. Is this going then to be pushed up to the app and have an update for the installed app that has this enabled?

**Reply by IAmTomahawkx** [reply]
The desktop client is simply a wrapper around our web app, so it will receive an update yes.

**Comment by TeejCodesTeej** [comment]
Push-to-talk is a desperately needed feature 🙏
